CDP4Reporting/DataCollection/CategoryDecompositionHierarchy.cs

```diff
@@ -26,130 +26,152 @@
 namespace CDP4Reporting.DataCollection
 {
     using System;
+    using System.Collections.Generic;
     using System.Linq;
 
     using CDP4Common.CommonData;
     using CDP4Common.EngineeringModelData;
     using CDP4Common.SiteDirectoryData;
 
     /// <summary>
-    /// Class representing a linear hierarchy of <see cref="CDP4Common.SiteDirectoryData.Category"/> items.
+    /// Class representing a linear hierarchy of <see cref="Category"/> items.
     /// </summary>
-    public class CategoryHierarchy
+    public class CategoryDecompositionHierarchy
     {
         /// <summary>
-        /// Builder class for a <see cref="CategoryHierarchy"/>.
+        /// Builder class for a <see cref="CategoryDecompositionHierarchy"/>.
         /// </summary>
         public class Builder
```

Comment:
any particular reason we have an class insice a class? we don't do that a lot, possible to split? and if not, does the visibility need to be `public`?

Reply:
Although this pattern was introduced by @cozminvelciu , I'm also used to having a builder inside its result class.
The main reason I use this for (Fluent) Builders is that the builder has access to the the result class' private properties, methods and constructors. 
Especially a private constructor is really nice, because then you protect your class so that it can only be instanciated using the Builder. That's also the reason the builder is public: It is used directly in the script editor.
I also think that the usage of the class is really nice and easy to understand.

Usage in script
```
var functionHierarchy = 
	new CategoryDecompositionHierarchy.Builder(this.Iteration)
		.AddLevel("Segments")
		.AddLevel("Systems")
		.AddLevel("Subsystems")
		.Build();
```

functionHierarchy is now a correctly built instance of CategoryDecompositionHierarchy.